### ISSUE TITLE
[Feature] Custom registration form

### ISSUE THREAD

**Issue by Quentinchampenois** [Platoniq/decidim-module-decidim_awesome#141]
### Description 
As discussed at the Decidim Fest we'd like to implement  a feature that allows admins to add fields on the decidim registration form. You'll find attached a spec document for this feature.

You suggested to use the same approach that used for custom proposal forms which uses [jQuery formBuilder library](https://formbuilder.online/) and allows to define HTML form inputs which you store as XML document in the Decidim DB.

We are a little concerned that this approach could have performance issues on the sign up form page which can be under heavier traffic since it's a common entry point for new users, which is not the case for the proposal form which is only accessible to logged in users. 

On the current sign up page we generate the form once, using jQuery formBuilder will add 2 steps : loading the js library and performing the modifications on the DOM. 

Before we started implementing this we researched another idea that came up before we came to the Decidim Fest which consist of using the decidim content blocks ([used in the homepage and process group pages](https://docs.decidim.org/en/develop/content_blocks/)). Which has the benefit of using something that's Decidim native. The drawback obviously is that we'd have to create a block by input type and might be too rigid to work with in the future.

For comparison we will try to implement the feature with form [https://formbuilder.online/](https://formbuilder.online/)  to see how it goes.

We thought we'd share our interrogations in here so we could you have your opinion on this. Thanks in advance and congrats on the awesome work 😉.

### Attached documents
* [Custom registration form.pdf](https://github.com/Platoniq/decidim-module-decidim_awesome/files/7547744/Custom.registration.form.pdf)

**Comment by virgile-dev** [comment]
@microstudi could you provide us feedback please ? Thanks in advance :)

**Comment by microstudi** [comment]
Hi  @virgile-dev @Quentinchampenois sorry for being unresponsive.

Your document looks indeed nice. This are my first impressions, I can be flexible 😁

- on one side, there's a contract from the Barcelona city council to be executed next year. Its about improving GDPR stuff and also adding under-age registration (with parental supervision). We are not sure how this going to affect registration/login process yet.
- on the performance concern: I don't see any problems the server-side wise, it's true though that user will have to activate javascript (although not sure how Decidim performs without it).
- Implementing formBuilder would give high flexibility, the challenge here is to design a powerful enough backend that would allow extract useful data/graphs. Possibly all this information could be encrypted in a JSON extra field of the user table, but that would make processing slow. Maybe this information could be in a different table non directly linked to the user to maximize anonymity. We should discuss the best way to do ti.

**Comment by Quentinchampenois** [comment]
Hi, thank you for sharing those points ! 

I agree with the need of flexibility, this is why we will probably implement this feature using FormBuilder and keeping in mind the need of anonymity.

**Comment by Quentinchampenois** [comment]
Hello Ivan,

Best wishes for 2022.
We worked on implementing the custom registration form feature based on your work for the custom proposal fields. We have implemented the JS FormBuilder as you did however it didn’t concluded successfully. Hence we’ll be closing this issue. 

For sure we’ll have other opportunities to  contribute on Decidim awesome however we think this feature might be too heavy to maintain in the future and add some complexity in the module. 

Product wise we realized that offering too many options for admins to configure their registration form might open the door to overly complicated forms for the users.

As a result,  we’ll working on a dedicated module which first version will feature predefined fields (age, gender, zip code).
